roles/openshift_storage_glusterfs/tasks/glusterfs_registry.yml

```diff
@@ -27,24 +27,13 @@
     glusterfs_heketi_ssh_user: "{{ openshift_storage_glusterfs_registry_heketi_ssh_user }}"
     glusterfs_heketi_ssh_sudo: "{{ openshift_storage_glusterfs_registry_heketi_ssh_sudo | bool }}"
     glusterfs_heketi_ssh_keyfile: "{{ openshift_storage_glusterfs_registry_heketi_ssh_keyfile }}"
-    glusterfs_nodes: "{{ groups.glusterfs_registry | default(groups.glusterfs) }}"
+    glusterfs_nodes: "{% if groups.glusterfs_registry is defined %}{% set nodes = groups.glusterfs_registry %}{% elif 'groups.glusterfs' is defined %}{% set nodes = groups.glusterfs %}{% else %}{% set nodes = '[]' %}{% endif %}{{ nodes }}"
```

Comment:
You don't need to use the interim `nodes` variable, just output the desired values between the `{% if %}` clauses.